app/parse.py

```
@@ -8,8 +15,51 @@ class Quote:
     tags: list[str]
 
 
+QUOTE_FIELDS = [field.name for field in fields(Quote)]
+
+
+def parse_single_quote(quote: Tag) -> Quote:
+    tags = [tag.text for tag in quote.select(".tag")]
+
+    return Quote(
+        text=quote.select_one(".text").text,
+        author=quote.select_one(".author").text,
+        tags=tags,
+    )
+
+
+def get_single_page_quotes(page_soup: Tag) -> list[Quote]:
+    quotes = page_soup.select(".quote")
+    return [parse_single_quote(quote) for quote in quotes]
+
+
+def parse_quotes() -> list[Quote]:
+    current_url = urljoin(QUOTES_URL, "/page/1/")
+    all_quotes = []
+    while True:
+        text = requests.get(current_url).content
```

You call `requests.get(current_url).content` without using a `requests.Session()` or specifying a `timeout`. Consider creating and reusing a `requests.Session()` and adding `timeout=` (e.g. `timeout=5`) or calling `response.raise_for_status()` to fail early on HTTP errors and to reuse TCP connections (friendlier and more robust).